src/SapoId/Provider.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace SocialiteProviders\SapoId;
+
+use GuzzleHttp\RequestOptions;
+use Laravel\Socialite\Two\ProviderInterface;
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+class Provider extends AbstractProvider implements ProviderInterface
+{
+    public const IDENTIFIER = 'SAPOID';
+    protected const ID_AUTH_URL = 'https://id.sapo.pt/oauth/v2/authorize';
+    protected const ID_TOKEN_URL = 'https://id.sapo.pt/oauth/v2/token';
+    protected const ID_USER_URL = 'https://id.sapo.pt/userinfo';
+
+    protected $scopeSeparator = ' ';
+
+    protected $scopes = ['openid'];
+
+    protected $usesPKCE = true;
+
+    public function setScopes($scopes)
+    {
+        $this->scopes = config('services.sapoid.scopes', $this->scopes);
```

Comment:
We should let them be set manually, not force from config as per other providers.
This won't let them be overridden.